modules/indexer/issue.go

```diff
@@ -116,17 +136,16 @@ func SearchIssuesByKeyword(repoID int64, keyword string) ([]int64, error) {
 		))
 	search := bleve.NewSearchRequestOptions(indexerQuery, 2147483647, 0, false)
 
-	result, err := issueIndexer.Search(search)
-	if err != nil {
-		return nil, err
-	}
+	return searchIssues(search)
+}
 
-	issueIDs := make([]int64, len(result.Hits))
-	for i, hit := range result.Hits {
-		issueIDs[i], err = idOfIndexerID(hit.ID)
-		if err != nil {
-			return nil, err
-		}
-	}
-	return issueIDs, nil
+// SearchIssuesByIndex searches for issues by given conditions.
+// Returns the matching issue IDs
+func SearchIssuesByIndex(repoID, issueIndex int64) ([]int64, error) {
```

Comment:
Why are you using the indexer? I think you should use the database instead.

In general, we only use indexers for text-based searches.

Reply:
Do you know good and effective solution for searching numbers like this?